src/model.jl

```diff
@@ -180,11 +186,27 @@ function _make_sets(case::Case, storage::Union{Storage,Nothing})::Sets
                                * "Did you forget to linearize_gencost?")
     num_segments = convert(Int, maximum(case.gencost[:, NCOST])) - 1
     segment_idx = 1:num_segments
+    # Demand flexibility, additional info for input->bus conversion
+    demand_flexibility_enabled = isa(demand_flexibility, DemandFlexibility) && demand_flexibility.enabled
```

Comment:
I'm not sure this is necessary. I believe `demand_flexibility` will always be of type `DemandFlexibility` (based on the call in `REISE.jl` and the definition in `read.jl`, even if there are no demand flexibility files present), so we shouldn't need the first check.

Reply:
understood. I will change it

Reply:
actually this seems to be required. I just recalled the reason I added this was in the end of solving optimization, the program calls make_sets() again without inputs, if I don't do isa() it will give an error.
> function _make_sets(case::Case)::Sets
>    _make_sets(case, nothing, nothing)
> end

Reply:
I still don't think it is required to be `nothing`, but I'm glad you pointed this out as it appears to have caught something we had missed. In the code snippet that you call out, the function is used in `query.jl` to create the sets that are needed for storing the results in a comprehensible way. Before this PR, we didn't have a `demand_flexibility` input because we just used `load_bus_map` when storing the bus-level demand flexibility deviations (see [here](https://github.com/dongqi-wu/REISE.jl-dqwu/blob/4bcb153a01169e10c82ebece81ea6bc9b77c10e1/src/query.jl#L72)); a `storage` input is also not needed because we don't use any of the storage-related sets there. Since the set of flexible load buses is no longer necessarily equal to the set of load buses, we will need to pass in `flexible_load_bus_map`, which is tied to `demand_flexibility`. So, we'll need to change the snippet that you shared above to become:

```
function _make_sets(case::Case, demand_flexibility::DemandFlexibility)::Sets
    _make_sets(case, nothing, demand_flexibility)
end
```

The code in `query.jl` will also need to be updated accordingly. We will also no longer need `nothing` to be a potential input type.

Reply:
@lanesmith According to l.147 it seems it can be `nothing` as well.

Reply:
@BainanXia I believe that was put there to enable the functionality in L.142-L.144. Since we need to pass `demand_flexibility` in that function, `nothing` is no longer needed as a potential input type (and should no longer need to be checked).

Reply:
@lanesmith I'm not familiar with Julia. What is l.142 - l.144 doing? It seems to me that it just adds a wrapper of `_make_sets` function.

Reply:
It appears to me that it is a way to create only a subset of the sets that are created in the full `_make_sets` function. This truncated version of `_make_sets` (specified in L.142-L.144) is (I believe) only used in `query.jl`, where the sets associated with `case` are used but the sets associated with `storage` are not used. This makes it so that some of the computations required to determine the storage-related sets are not performed, which should make it faster (though I'm not sure how much faster). @danielolsen could probably give a better answer as to whether or not the function specified in L.142-L.144 is doing more.

Having looked at this a little more, I wonder if we would be able to create `sets` once and take advantage of our `Sets` struct. Could it be called once in `REISE.jl`?

Reply:
But both functions are defined in the same domain with same name, will the full `_make_sets` overrides the truncated one?

Reply:
This is where we approach the end of my limited Julia knowledge haha. Doing a simple test (and based on the functions working in our simulations), it appears that Julia can tell the difference between tests of the same name so long as they have different inputs. I'm not certain that we can do this in Julia, but if we were to do this in a more Pythonic way, we could probably set `storage=nothing` as a default in the main `_make_sets` function.